crosshair/libimpl/builtinslib.py

```diff
@@ -1916,6 +1916,31 @@ def find(self, substr, start=None, end=None):
         start = 0 if start is None else force_to_smt_sort(start, SmtInt)
         return SmtInt(z3.IndexOf(smt_mystr, smt_substr, start))
 
+    def rfind(self, substr, start=None, end=None):
+        space = self.statespace
+        sub = force_to_smt_sort(substr, SmtStr)
+
+        start = 0 if start is None else force_to_smt_sort(start, SmtInt)
+        end = z3.Length(value) if end is None else force_to_smt_sort(end, SmtInt)
+        value = z3.SubString(self.var, start, end - 1)
+        match_index = z3.Int("match_index")
+        return_value = z3.Int("return_value")
```

Comment:
```suggestion
        match_index = z3.Int("match_index" + space.uniq())
        return_value = z3.Int("return_value" + space.uniq())
```
Different values need to have different names in Z3. This just adds an underscore and some unique hex value to ensure it's unique. (otherwise you'll run into confusing behavior when calling rfind repeatedly!)

Reply:
This was my hypothesis this morning when I saw the strange test results. I've made the change, and the test now passes! Added in latest commit. :)